dist/profile/manifests/datadog_http_check.pp

```diff
@@ -0,0 +1,171 @@
+# Class: profile::datadog_http_check
+#
+# This class will install the necessary config to monitor jenkins infrastructure https endpoint
+# Parameters:
+#     contact
+#       Specify a list of users to notify regarding http_check alert
+#
+#     days_critical
+#     days_warning
+#        The check_certificate_expiration will instruct the check
+#        to create a service check that checks the expiration of the
+#        ssl certificate. Allow for a warning to occur when x days are
+#        left in the certificate, and alternatively raise a critical
+#        warning if the certificate is y days from the expiration date.
+#        The SSL certificate will always be validated for this additional
+#         service check regardless of the value of disable_ssl_validation
+#
+#     timeout
+#        The (optional) timeout in seconds.
+#
+#     threshold
+#     window
+#        The (optional) window and threshold parameters allow you to trigger
+#        alerts only if the check fails x times within the last y attempts
+#        where x is the threshold and y is the window.
+#
+
+class profile::datadog_http_check(
+  $contact = ['pagerduty'],
+  $days_warning = 30,
+  $days_critical = 10,
+  $timeout = 1,
+  $treshold = 3,
+  $window = 5
+){
+
+  include datadog_agent
+
+  class { 'datadog_agent::integrations::http_check':
```

Comment:
I was already concerned by that but my puppetfu is not good enough to solve it.
And I don't think that 'exported resources' will help here.
We can only call the class 'datadog_agent::integrations::http_check' one time per node
  
'profile::datadog_http_check' use the variable 'profile::datadog_http_check::instances'
to know what need to be define in /etc/dd-agent/conf.d/http_check.yaml.

Which means that each application needs to append his instance configuration to profile::datadog_http_check::instances.

A solution would be to move this variable to hiera